astropy/io/votable/tree.py

```diff
@@ -3509,6 +3510,32 @@ def iter_groups(self):
     def iter_info(self):
         yield from self.infos
 
+    def pop(self, index=-1):
+        """
+        Remove and return item at index (default last).
+
+        Raises IndexError if list is empty or index is out of range.
+        """
+        # Ensure we're within bounds of the array
+        if abs(index) > len(self.array.data):
+            raise IndexError("pop index out of range")
+        # If negative indexing, convert to true index
+        elif index < 0:
+            index = len(self.array) + index
```

Comment:
Maybe the index check should happen after converting to "true index"? Otherwise, negative index is never checked?

I think we can also special case the situation when the table only has one row and index given is 0. You basically return an empty table without having to use numpy at all. Maybe.